src/async/DataConsistencyRealmStore.cc

```diff
@@ -0,0 +1,26 @@
+// SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+#include "async/DataConsistencyRealmStore.h"
+
+namespace ChimeraTK::async {
+
+  /********************************************************************************************************************/
+
+  std::shared_ptr<DataConsistencyRealm> DataConsistencyRealmStore::getRealm(const std::string& realmName) {
+    std::lock_guard<std::mutex> lock(_mapMutex);
+
+    auto it = _realmMap.find(realmName);
```

Comment:
This looks "meh" and somehow should be shorter with try_emplace but probably can't because of the weak_ptr

Reply:
The other problem is that with `try_emplace()` that we can only create a weak_ptr from an existing shared_ptr. Hence we would always create a new realm first (to have a shared_ptr), and then throw it away again if it was already on the map. I think we have to keep it as is.